File: datamodel/src/main/java/datamodel/clinical/ClinicalStatus.java
package datamodel.clinical;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class ClinicalStatus {

    @Nullable
    public abstract Integer who();

    @Nullable
    public abstract InfectionStatus infectionStatus();

    @Nullable
    public abstract ECG ecg();

    @Nullable
    public abstract Double lvef();

    @Nullable
    public abstract Boolean hasComplications();
}

File: algo/src/main/kotlin/com/hartwig/actin/datamodel/clinical/ClinicalStatus.kt
package com.hartwig.actin.datamodel.clinical

data class ClinicalStatus(
    val who: Int? = null,
    val infectionStatus: InfectionStatus? = null,
    val ecg: Ecg? = null,
    val lvef: Double? = null,
    val hasComplications: Boolean? = null
)
